compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirExpressionsResolveTransformer.kt

```diff
@@ -187,9 +188,46 @@ class FirExpressionsResolveTransformer(transformer: FirBodyResolveTransformer) :
             }
 
         dataFlowAnalyzer.exitFunctionCall(completeInference, callCompleted)
+        if (callCompleted && completeInference.isArrayOfCall) {
```

Comment:
It seems to me that the following transformation should be done only if we are resolving annotation arguments now. Not quite sure how good or bad is the transformation of a regular `arrayOf` call (not inside an annotation) to `FirArrayOfCall`.

Reply:
Make sense. Added one more commit that migrates transformation-related logic into a separate transformer, then tweak expression transformer a bit to use it only if we're resolving annotation call or default value of annotation constructor (as per two unmuted tests).

Reply:
Btw, feel free to squash last two commits if you think the intro of `FirArrayOfCallTransformer` and its application to expression transformer should stick together. And, that will remove unnecessary nosies of commit changes in those IR text files. But, hope the separate one is helpful for you to review.